contracts/contracts/token/OUSD.sol

```diff
@@ -32,40 +27,61 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
     );
     event AccountRebasingEnabled(address account);
     event AccountRebasingDisabled(address account);
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(
+        address indexed owner,
+        address indexed spender,
+        uint256 value
+    );
 
     enum RebaseOptions {
         NotSet,
-        OptOut,
-        OptIn
+        StdNonRebasing,
+        StdRebasing,
+        YieldDelegationSource,
+        YieldDelegationTarget
     }
 
+    // Add slots to align with deployed OUSD contract
+    uint256[154] private _gap;
     uint256 private constant MAX_SUPPLY = ~uint128(0); // (2^128) - 1
     uint256 public _totalSupply;
     mapping(address => mapping(address => uint256)) private _allowances;
     address public vaultAddress = address(0);
-    mapping(address => uint256) private _creditBalances;
-    uint256 private _rebasingCredits;
+    mapping(address => uint256) internal _creditBalances;
+    uint256 private _rebasingCredits; // Sum of all rebasing credits (_creditBalances for rebasing accounts)
     uint256 private _rebasingCreditsPerToken;
-    // Frozen address/credits are non rebasing (value is held in contracts which
-    // do not receive yield unless they explicitly opt in)
-    uint256 public nonRebasingSupply;
-    mapping(address => uint256) public nonRebasingCreditsPerToken;
+    uint256 public nonRebasingSupply; // All nonrebasing balances
+    mapping(address => uint256) internal alternativeCreditsPerToken;
     mapping(address => RebaseOptions) public rebaseState;
     mapping(address => uint256) public isUpgraded;
+    mapping(address => address) public yieldTo;
+    mapping(address => address) public yieldFrom;
 
     uint256 private constant RESOLUTION_INCREASE = 1e9;
 
-    function initialize(
-        string calldata _nameArg,
-        string calldata _symbolArg,
-        address _vaultAddress,
-        uint256 _initialCreditsPerToken
-    ) external onlyGovernor initializer {
-        InitializableERC20Detailed._initialize(_nameArg, _symbolArg, 18);
+    function initialize(address _vaultAddress, uint256 _initialCreditsPerToken)
+        external
+        onlyGovernor
+    {
+        require(vaultAddress == address(0), "Already initialized");
+        require(_rebasingCreditsPerToken == 0, "Already initialized");
```

Comment:
We don't need to check both of these. Instead we should check `vaultAddress` as we currently do then, check that the new `_vaultAddress` is nonzero. In this way the function can only be called once.

Reply:
done here: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/2b31fd654bdb9fbd4536badbbb1c7e78433f59a8